src/classlibnative/bcltype/system.cpp

```diff
@@ -225,17 +225,18 @@ FCIMPL1(Object*, SystemNative::_GetEnvironmentVariable, StringObject* strVarUNSA
     int len, newLen;
 
     // Get the length of the environment variable.
-    WCHAR dummy;    // prefix complains if pass a null ptr in, so rely on the final length parm instead
-    len = WszGetEnvironmentVariable(strVar->GetBuffer(), &dummy, 0);
+    PathString dummy;    // prefix complains if pass a null ptr in, so rely on the final length parm instead
+    len = WszGetEnvironmentVariable(strVar->GetBuffer(), dummy);
 
     while (len != 0)
     {
         // Allocate the string.
-        refRetVal = StringObject::NewString(len);
-
-        // Get the value.
-        newLen = WszGetEnvironmentVariable(strVar->GetBuffer(), refRetVal->GetBuffer(), len);
-        if (newLen != (len - 1))
+        refRetVal = StringObject::NewString(len + 1);
```

Comment:
It looks like we have a long standing off-by-one bug here. StringObject::NewString takes the length of the string, without counting the null terminator. So we should pass len-1 before the change, and len after the change.